src/routinesWatcherSaga.js

```diff
@@ -1,19 +1,23 @@
-import { takeEvery, take, race, put, call } from 'redux-saga/effects';
+import { takeEvery, take, race, put, call, all } from 'redux-saga/effects';
 import { PROMISE_ACTION } from './constants'
 
 const getPayload = (data) => (data && data.payload) || data;
 
 export function* handlePromiseAction(action) {
   const { data, params, defer: { resolve, reject } } = action.payload;
 
-  const [ { success, failure } ] = yield [
+  const effects = [
     race({
       success: take(params.SUCCESS),
       failure: take(params.FAILURE),
     }),
     put(params.trigger(data)),
   ];
 
+  const allEffects = all([...effects]).ALL;
+
+  const [ {success, failure} ] = yield allEffects;
```

Comment:
I did not found `.ALL` in `redux-saga` docs. I think following code is more compact and easy to understand:
```javascript
const [ { success, failure } ] = yield all([
  race(...),
  put(...),
]);
```

Reply:
![Here is the test result with your suggestion.](https://user-images.githubusercontent.com/18321894/30950632-04a2c13c-a3d3-11e7-846c-c9b9ad1c1d92.png)

![Here is the test result with '.ALL' as I implemented it in this PR.](https://user-images.githubusercontent.com/18321894/30950692-59be8534-a3d3-11e7-93d9-aa8706f97ab5.png)

Yes, I understand there is no '.ALL' in the redux saga documentation, but I for the life of me cannot figure out how to get the tests to pass without it as the tests already stand, and I haven't been able to figure out how to fix the tests otherwise.

This bug is *extremely* cumbersome on my console. Let's see if we can get it cleared up soon 👍